vscode/src/chat/chat-view/SimpleChatPanelProvider.ts

```diff
@@ -108,10 +107,38 @@ export interface ChatSession {
     sessionID: string
 }
 
+/**
+ * SimpleChatPanelProvider is the view controller class for the chat panel.
+ * It handles all events sent from the view, keeps track of the underlying chat model,
+ * and interacts with the rest of the extension.
+ *
+ * Its methods are grouped into the following sections, each of which is demarcated
+ * by a comment block (search for "// Section: "):
+ *
+ * 1. top-level view action handlers
+ * 2. view updaters
+ * 3. chat request lifecycle methods
+ * 4. session management
+ * 5. webview container management
+ * 6. other public accessors and mutators
+ *
+ * The following invariants should be maintained:
+ * 1. top-level view action handlers
+ *    a. should all follow the handle$ACTION naming convention
+ *    b. should be private (with the existing exceptions)
+ * 2 view updaters
+ *    a. should all follow the post$ACTION naming convention
+ *    b. should NOT mutate model state
+ * 3. Do NOT introduce new public fields or methods. The public
+ *    interface of this class should be kept small in order to
+ *    avoid tight coupling with other classes. If communication
+ *    with other components outside the model and view is needed,
+ *    we should implement an broadcast/subscription design.
```

Comment:
Focus on the positive of what *to* do?

```suggestion
 * 3. Keep the public
 *    interface of this class should be small in order to
 *    avoid tight coupling with other classes. If communication
 *    with other components outside the model and view is needed,
 *    use a broadcast/subscription design.
```